cmake/reports/CMakeLists.txt

```
@@ -48,22 +49,23 @@ if (CONFIG_BUILD_WITH_TFM)
   endforeach()
 endif()
 
-if (CONFIG_TFM_BL2)
+if(CONFIG_TFM_BL2)
   foreach(report ram_report rom_report footprint)
-  add_custom_target(
-    bl2_${report}
-    ${PYTHON_EXECUTABLE}
-    ${ZEPHYR_BASE}/scripts/footprint/size_report
-    -k $<TARGET_PROPERTY:tfm,BL2_ELF_FILE>
-    -z ${ZEPHYR_BASE}
-    -o ${CMAKE_BINARY_DIR}
-    ${workspace_arg}
-    -d ${report_depth}
-    ${flag_for_${report}}
-    DEPENDS tfm
-    USES_TERMINAL
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-    )
+    add_custom_target(
+      bl2_${report}
+      ${PYTHON_EXECUTABLE}
+      ${ZEPHYR_BASE}/scripts/footprint/size_report
+      -k $<TARGET_PROPERTY:tfm,BL2_ELF_FILE>
+      -z ${ZEPHYR_BASE}
+      -o ${CMAKE_BINARY_DIR}
+      ${workspace_arg}
+      -d ${report_depth}
+      --prefix bl2
```

`size_report` already supports writing to an alternative file using `--json <file>`, so why not use what we have instead of introducing yet another argument ?

```suggestion
      --json bl2_${report}.json
```

---

much simpler solution... Applied